trace_replay/block_cache_tracer.h

```diff
@@ -0,0 +1,92 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#pragma once
+
+#include "monitoring/instrumented_mutex.h"
+#include "rocksdb/env.h"
+#include "rocksdb/options.h"
+#include "rocksdb/trace_reader_writer.h"
+#include "trace_replay/trace_replay.h"
+
+namespace rocksdb {
+
+enum BlockCacheLookupCaller : char {
+  kUserGet = 1,
+  kUserMGet = 2,
+  kUserIterator = 3,
+  kPrefetch = 4,
+  kCompaction = 5,
+  // All callers should be added before kMaxBlockCacheLookupCaller.
+  kMaxBlockCacheLookupCaller
+};
+
+enum Boolean : char { kTrue = 1, kFalse = 0 };
+
+struct BlockCacheTraceRecord {
+  uint64_t access_timestamp;
+  std::string block_key;
+  TraceType block_type;
+  uint64_t block_size;
+  uint32_t cf_id;
+  std::string cf_name;
+  uint32_t level;
+  uint32_t sst_fd_number;
+  BlockCacheLookupCaller caller;
+  Boolean is_cache_hit;
+  Boolean no_insert;
+
+  // For Data block and User Get/Multi-Get only.
+  std::string referenced_key;
+  uint64_t num_keys_in_block = 0;
+  Boolean is_referenced_key_exist_in_block = Boolean::kFalse;
```

Comment:
Any reason why only a subset of the struct's members are initialized?

Reply:
These fields are required only for data block and user get/multi-get. The test checks that their values equal to their default values for other types of accesses.